### PR TITLE
C# code generator improvements

### PR DESCRIPTION
**Pull request by cezarypiatek**
This PR brings the following improvements:
- Add quote escaping in generated C# strings
- Handle JSON in the response body. Json is converted to C# anonymous object definition and fed into `WithBodyAsJson` method.

### CODE REVIEW

File: src/WireMock.Net/Serialization/MappingConverter.cs [hunk 1]
```diff
@@ -173,6 +187,49 @@ public string ToCSharpCode(IMapping mapping, MappingConverterSettings? settings
         return sb.ToString();
     }
 
+    private string ConvertJsonToAnonymousObjectDefinition(JToken token, int ind = 0)
+    {
+        string FormatObject(JObject jObject)
+        {
+            var indStr = new string(' ', 4 * ind);
+            var indStrSub = new string(' ', 4 * (ind+1));
```
Comment (StefH) [inline src/WireMock.Net/Serialization/MappingConverter.cs:195]: can you do a format document in visual studio?

File: src/WireMock.Net/Serialization/MappingConverter.cs [hunk 2]
```diff
@@ -173,6 +187,49 @@ public string ToCSharpCode(IMapping mapping, MappingConverterSettings? settings
         return sb.ToString();
     }
 
+    private string ConvertJsonToAnonymousObjectDefinition(JToken token, int ind = 0)
```
Comment (StefH) [inline src/WireMock.Net/Serialization/MappingConverter.cs:190]: Another possible option would be to just use reflection:
``` c#
private static string GenerateObjectInitializationCodeRecursive(object anonymousObject)
{
	var properties = anonymousObject.GetType().GetProperties();
	var objectBuilder = new StringBuilder();

	objectBuilder.AppendLine("new");
	objectBuilder.AppendLine("{");

	foreach (var property in properties)
	{
		string propertyName = property.Name;
		object propertyValue = property.GetValue(anonymousObject);

		if (propertyValue != null)
		{
			if (propertyValue is string)
			{
				objectBuilder.AppendLine($"    {propertyName} = \"{propertyValue}\",");
			}
			else if (property.PropertyType.IsClass && !property.PropertyType.FullName.StartsWith("System."))
			{
				string nestedObjectCode = GenerateObjectInitializationCodeRecursive(propertyValue);
				objectBuilder.AppendLine($"    {propertyName} = {nestedObjectCode.Trim()},");
			}
			else
			{
				objectBuilder.AppendLine($"    {propertyName} = {propertyValue},");
			}
		}
	}

	objectBuilder.AppendLine("};");

	return objectBuilder.ToString();
}
```

**Review by StefH — CHANGES_REQUESTED** [review]

File: src/WireMock.Net/Serialization/MappingConverter.cs (hunk 1 above)
Comment (cezarypiatek) [reply src/WireMock.Net/Serialization/MappingConverter.cs]: Fixed

File: src/WireMock.Net/Serialization/MappingConverter.cs (hunk 2 above)
Comment (cezarypiatek) [reply src/WireMock.Net/Serialization/MappingConverter.cs:190]: Yes, it's quite common that a given problem can be solved in multiple ways. Try to use the code from above and there are compilation issues (probably due to missing properties in some target frameworks) and I guess some additional adjustments are needed to achieve proper formatting.

Is there any reason why Newtonsoft.Json can't be used here? It's already referenced by WireMock project.

Comment (StefH) [reply src/WireMock.Net/Serialization/MappingConverter.cs:190]: Your solution is fine. It was just an idea.

**Review by StefH — APPROVED** [review]

**Comment by cezarypiatek** [comment]
Thanks for merging. Btw, do you have a plan for the next release? I would like to start using the code generation feature but I need those recent changes.

**Comment by StefH** [comment]
@cezarypiatek Today I'll release new NuGet.

**Pull request merged**